streams/src/test/java/org/apache/kafka/streams/integration/KafkaStreamsTelemetryIntegrationTest.java

```diff
@@ -0,0 +1,547 @@
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        INTERCEPTING_CONSUMERS.clear();
+        INTERCEPTING_ADMIN_CLIENTS.clear();
+        IntegrationTestUtils.purgeLocalStreamsState(streamsConfigurations);
+        streamsConfigurations.clear();
+    }
+
+    @Test
+    @DisplayName("Calling unregisterMetric on metrics not registered should not cause an error")
+    public void shouldNotThrowExceptionWhenRemovingNonExistingMetrics() throws InterruptedException {
+        final Properties properties = props(true);
+        final Topology topology = complexTopology();
+        try (final KafkaStreams streams = new KafkaStreams(topology, properties)) {
+            streams.start();
+            waitForCondition(() -> KafkaStreams.State.RUNNING == streams.state(),
+                    IntegrationTestUtils.DEFAULT_TIMEOUT,
+                    () -> "Kafka Streams never transitioned to a RUNNING state.");
+
+            final Consumer<?, ?> embeddedConsumer = INTERCEPTING_CONSUMERS.get(FIRST_INSTANCE_CONSUMER);
+            final MetricName metricName = new MetricName("fakeMetric", "fakeGroup", "It's a fake metric", new HashMap<>());
+            final KafkaMetric nonExitingMetric = new KafkaMetric(new Object(), metricName, (Measurable) (m, now) -> 1.0, new MetricConfig(), Time.SYSTEM);
+            assertDoesNotThrow(() -> embeddedConsumer.unregisterMetricFromSubscription(nonExitingMetric));
+        }
+    }
+
+    @Test
+    @DisplayName("End-to-end test validating metrics pushed to broker")
+    public void shouldPushMetricsToBroker() throws Exception {
+        final Properties properties = props(true);
+        final Topology topology = simpleTopology();
+        subscribeForStreamsMetrics();
+        try (final KafkaStreams streams = new KafkaStreams(topology, properties)) {
+            IntegrationTestUtils.startApplicationAndWaitUntilRunning(streams);
+            final ClientInstanceIds clientInstanceIds = streams.clientInstanceIds(Duration.ofSeconds(60));
+            final Uuid adminInstanceId = clientInstanceIds.adminInstanceId();
+            final Uuid mainConsumerInstanceId = clientInstanceIds.consumerInstanceIds().entrySet().stream()
+                    .filter(entry -> !entry.getKey().contains("restore"))
+                    .map(Map.Entry::getValue)
+                    .findFirst().get();
+            assertNotNull(adminInstanceId);
+            assertNotNull(mainConsumerInstanceId);
+
+            TestUtils.waitForCondition(() -> !SUBSCRIBED_CLIENT_METRICS.get(mainConsumerInstanceId).isEmpty(),
+                    30_000,
+                    "Never received subscribed metrics");
+            final List<String> actualTaskMetrics = SUBSCRIBED_CLIENT_METRICS.get(mainConsumerInstanceId).stream().filter(metricName -> metricName.startsWith("org.apache.kafka.stream.task")).collect(Collectors.toList());
+            assertEquals(EXPECTED_MAIN_CONSUMER_TASK_METRICS, actualTaskMetrics);
```

Comment:
Picking grabbing task metrics as a good indicator that streams metrics push is working.

Reply:
Given the test name, can't we access the broker side plugin to verify what metrics got pushed? Should we verify instance-level, ie, `admin`, too?

Reply:
> Given the test name, can't we access the broker side plugin to verify what metrics got pushed?

That's exactly what the test does, the Map containing the metrics name is populated in the plugin but this may have been obscured a bit by naming - I've updated the code to make it more clear what's going on